src/foam/nanos/approval/ApprovalRequest.js

```diff
@@ -0,0 +1,440 @@
+      If "approver" property is set, "group" property is ignored.`,
+      section: 'supportDetails',
+      visibility: function(group) {
+        return group ?
+          foam.u2.DisplayMode.RO :
+          foam.u2.DisplayMode.HIDDEN;
+      }
+    },
+    {
+      class: 'Enum',
+      of: 'foam.nanos.approval.ApprovalStatus',
+      name: 'status',
+      value: 'REQUESTED',
+      section: 'requestDetails',
+      javaFactory: 'return foam.nanos.approval.ApprovalStatus.REQUESTED;',
+      visibility: function(status) {
+        return status ?
+          foam.u2.DisplayMode.RO :
+          foam.u2.DisplayMode.HIDDEN;
+      }
+    },
+    {
+      class: 'String',
+      name: 'memo',
+      view: { class: 'foam.u2.tag.TextArea', rows: 5, cols: 80 },
+      documentation: 'Meant to be used for explanation on why request was approved/rejected',
+      section: 'basicInformation',
+      visibility: function(memo) {
+        return memo ?
+          foam.u2.DisplayMode.RO :
+          foam.u2.DisplayMode.HIDDEN;
+      }
+    },
+    {
+      class: 'String',
+      name: 'description',
+      documentation: `Approval request description.`,
+      tableWidth: 200,
+      section: 'basicInformation',
+      visibility: function(description) {
+        return description ?
+          foam.u2.DisplayMode.RO :
+          foam.u2.DisplayMode.HIDDEN;
+      }
+   },
+    {
+      class: 'String',
+      name: 'token',
+      documentation: 'token in email for ‘click to approve’.',
+      section: 'basicInformation',
+      visibility: 'HIDDEN'
+    },
+    {
+      class: 'DateTime',
+      name: 'created',
+      section: 'supportDetails',
+      gridColumns: 6,
+      visibility: function(created) {
+        return created ?
+          foam.u2.DisplayMode.RO :
+          foam.u2.DisplayMode.HIDDEN;
+      }
+    },
+    {
+      class: 'DateTime',
+      name: 'lastModified',
+      gridColumns: 6,
+      section: 'supportDetails',
+      visibility: function(lastModified) {
+        return lastModified ?
+          foam.u2.DisplayMode.RO :
+          foam.u2.DisplayMode.HIDDEN;
+      }
+    },
+    {
+      class: 'Reference',
+      of: 'foam.nanos.auth.User',
+      name: 'createdBy',
+      section: 'supportDetails',
+      visibility: 'HIDDEN'
+    },
+    {
+      class: 'Reference',
+      of: 'foam.nanos.auth.User',
+      name: 'createdByAgent',
+      section: 'supportDetails',
+      visibility: 'HIDDEN'
+    },
+    {
+      class: 'Reference',
+      of: 'foam.nanos.auth.User',
+      name: 'lastModifiedBy',
+      section: 'supportDetails',
+      visibility: 'HIDDEN'
+    },
+    {
+      class: 'String',
+      name: 'refObj',
+      transient: true,
+      expression: function(daoKey, objId) {
+        return daoKey + ':' + objId;
+      },
+      javaGetter: `
+        return getDaoKey() + ": " + getObjId();
+      `,
+      hidden: true
+    },
+    {
+      class: 'String',
+      name: 'daoKey_',
+      hidden: true,
```

Comment:
visibility please.

Reply:
Was last minute fix, just changed it to permissionRequired: true

Reply:
Was last minute fix, just changed it to permissionRequired: true